File: integration-tests/TestAPI/bindings/c-sharp/Cricut.TestAPI.Tests/URLTests.cs
using System;
using Xunit;
using Cricut.FishyJoesRuntime;

namespace Cricut.TestAPI.Tests {
    [Collection("root")]
    public class URLTests {
        [Fact]
        void testURLValues() {
            Assert.Equal(new Uri("https://www.google.com"), URLs.Simple);
            Assert.Equal(new Uri("https://www.google.com/images/branding/googlelogo/2x/googlelogo_light_color_272x92dp.png"), URLs.RemoteFile);
            Assert.Equal(new Uri("file:///Users/someuser/Desktop/a.png"), URLs.LocalFile);
        }

        [Fact]
        void testURLEcho() {
            Assert.Equal(URLs.Simple, URLs.Echo(URLs.Simple));
            Assert.Equal(URLs.RemoteFile, URLs.Echo(URLs.RemoteFile));
            Assert.Equal(URLs.LocalFile, URLs.Echo(URLs.LocalFile));
        }
    }
}

File: integration-tests/TestAPI/bindings/kotlin/src/test/kotlin/com/cricut/testapi/URLTests.kt
package com.cricut.testapi

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test
import java.net.MalformedURLException
import java.net.URI

class URLTests {
    @Test
    fun testURLValues() {
        assertEquals(URI("https://www.google.com"), URLs.simple)
        assertEquals(URI("https://www.google.com/images/branding/googlelogo/2x/googlelogo_light_color_272x92dp.png"), URLs.remoteFile, )
        assertEquals(URI("file:///Users/someuser/Desktop/a.png"), URLs.localFile)
        assertEquals("data", URLs.dataImage.scheme)
    }

    @Test
    fun testURLEcho() {
        assertEquals(URLs.simple, URLs.echo(URLs.simple))
        assertEquals(URLs.remoteFile, URLs.echo(URLs.remoteFile))
        assertEquals(URLs.localFile, URLs.echo(URLs.localFile))
        assertEquals(URLs.dataImage, URLs.echo(URLs.dataImage))
    }
}
